tests/test_gradient.py

```diff
@@ -9,31 +9,20 @@
 
 class TestGradient(object):
 
-    @pytest.mark.parametrize('space_order', [4])
-    @pytest.mark.parametrize('kernel', ['OT2'])
-    @pytest.mark.parametrize('shape', [(70, 80)])
-    def test_gradient_checkpointing(self, shape, kernel, space_order):
+    @pytest.mark.parametrize('dtype, opt', [
+        (np.float32, ('noop', {'openmp': True, 'par-dynamic-work': 1000})),
```

Comment:
`par-dynamic-work` ? that's Navjot's hack around the original failure. Unnecessary here?

Reply:
No, it's still necessary in single-precision since dynamic scheduling leads to non-bitwise reproducibility even with correct generated code.

Reply:
Is that really true - I mean have you run it and had it fail? I've been running this thousands of times on two different systems and never once an issue.

Reply:
Maybe not, haven't tried latest versio

Reply:
@ggorman yep still neede, fails without it.

Reply:
BTW, with @tjb900 we recently talked about implementing deterministic injection for the SparseFunction submodule inside dugwave

Reply:
I rewrote this so that only the gradient Function is double (np.float64) to avoid the accumulation of roundoff errors after thousands of interactions. No more noop.

Reply:
That's not good, can we revert to previous setup. Forcing the gradient to `np.float64` is basically bypassing the tst setup to do compute n double precision so we only pretend to have the single precision test passing.

Reply:
I'm not sure why was this conversation marked as resolved when it's clearly not. Also, we're still disabling dynamic parallelism (option `par-dynamic-work` ) for no reason)